frontend/public/components/marketplace/kubernetes-marketplace-items.jsx

```diff
@@ -151,62 +165,145 @@ export class MarketplaceTileViewPage extends React.Component {
     return items;
   }
 
-  getFilterCountsFromGroups(items, filters, itemsFilteredByGroup) {
-    // Input items, filters, and mapping from filter property to filtered items
-    // Returns object mapping filter property to item counts
-    return _.reduce(filters, (counts, group, key) => {
-      if (key === 'keyword') {
-        return counts;
-      }
+  // This function is necessary due to calls to history.replace un-mounting the component before returning
+  updateMountedState(newState) {
+    if (!this.unmounted) {
+      this.setState(newState);
```

Comment:
We have a `SafetyFirst` class that does something like this, although @alecmerdler is trying to refactor it to be a HOC.

Do we know under what circumstances this happens? I wouldn't expect to need this check unless we are using timeouts, XHRs, resize listeners, etc. (The problem I noticed in @dtaylor113's PR was apparently the one you fixed in the upstream resize listener.)